portal-api/src/v1/gef/services/clone/helpers/generate-cloned-deal.test.ts

```diff
@@ -0,0 +1,131 @@
+import { aPortalSessionUser, Bank, Deal, DEAL_STATUS, getCurrentGefDealVersion } from '@ukef/dtfs2-common';
+import { ObjectId } from 'mongodb';
+import { generateAuditDatabaseRecordFromAuditDetails, generateSystemAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { generateClonedExporter } from './generate-cloned-exporter';
+import { generateClonedDeal } from './generate-cloned-deal';
+
+const mockLatestEligibilityCriteria = {
+  eligibility: 'criteria',
+};
+
+const existingDealId = new ObjectId().toString();
+const mockAuditDetails = generateSystemAuditDetails();
+const mockBank = {
+  id: 'bank id',
+} as Bank;
+const bankInternalRefName = 'bankInternalRefName';
+const additionalRefName = 'additionalRefName';
+const maker = aPortalSessionUser();
+
+const existingDeal: Deal = {
+  _id: new ObjectId(existingDealId),
+  version: 0,
+  dealType: 'GEF',
+  ukefDealId: 'UkefDealId',
+  eligibility: {},
+  exporter: {},
+  createdAt: Date.now(),
+  updatedAt: Date.now(),
+  facilitiesUpdated: Date.now(),
+  status: DEAL_STATUS.UKEF_ACKNOWLEDGED,
+  submissionType: null,
+  submissionDate: null,
+  submissionCount: 0,
+  bankInternalRefName: 'oldBankInternalRefName',
+  additionalRefName: 'oldAdditionalRefName',
+  maker: {},
+  bank: {},
+  checkerId: 'checker id',
+  editedBy: ['a user id', 'another user id'],
+  portalActivities: ['An activity'],
+  supportingInformation: {
+    information: 'information',
+  },
+  auditRecord: {},
+  ukefDecision: 'ukefDecision',
+  ukefDecisionAccepted: 'ukefDecisionAccepted',
+  checkerMIN: 'checkerMIN',
+  manualInclusionNoticeSubmissionDate: 'manualInclusionNoticeSubmissionDate',
+  comments: 'comments',
+  previousStatus: 'previousStatus',
+  dataMigration: 'dataMigration',
```

Comment:
Looks like this should be an object?

Reply:
I dont think these types actually matter (I've taken a similar attitude for all the values that are deleted) since its a unit test. but will update this